game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/sub/SubmergeSubsVsOnlyAirStep.java

```diff
@@ -0,0 +1,72 @@
+package games.strategy.triplea.delegate.battle.steps.retreat.sub;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SUBMERGE_SUBS_VS_AIR_ONLY;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import org.triplea.java.collections.CollectionUtils;
+
+/** Units that canNotBeTargetedByAll can submerge if there are only Air units in the battle */
+public class SubmergeSubsVsOnlyAirStep extends BattleStep {
+
+  private static final Predicate<Unit> canNotBeTargetedByAllMatch =
+      Matches.unitCanEvade().and(Matches.unitCanNotBeTargetedByAll());
+
+  public SubmergeSubsVsOnlyAirStep(
+      final BattleState battleState, final BattleActions battleActions) {
+    super(battleState, battleActions);
+  }
+
+  @Override
+  public BattleAtomic getExecutable() {
+    return new BattleAtomic() {
+      private static final long serialVersionUID = 99990L;
+    };
+  }
+
+  @Override
+  public List<String> getNames() {
+    return List.of(SUBMERGE_SUBS_VS_AIR_ONLY);
```

Comment:
If we always return 0 or 1 element, would `Optional<String>` be a better return API? In that case there would not be any confusion that we can have more than one element and it'd be clear we'd have either zero or one.

Reply:
No, there are steps that will have more than 1.  An example is the step that fires aa guns.  The step will return "attacker fires guns", "defender selects casualties", and "defender removes casualties".

Now, it is possible after further refactoring, that step could be split up into three separate steps.  But I'm not sure at this point.